lib/db.c

```diff
@@ -1614,6 +1622,22 @@ grn_table_add(grn_ctx *ctx, grn_obj *table, const void *key, unsigned int key_si
           } GRN_TABLE_LOCK_END(ctx, table);
         });
         if (added) { *added = added_; }
+        if (id == GRN_ID_NIL) {
+          grn_obj buffer;
+          GRN_TEXT_INIT(&buffer, 0);
+          grn_inspect(ctx, &buffer, table);
+          if (key && key_size > 0) {
+            GRN_LOG(ctx, GRN_LOG_ERROR,
+                    "[table][add] failed to add key for TABLE_PAT_KEY: <%.*s> key: <%.*s>",
```

Comment:
`failed to add` is enough. We can only add key for tables. Inspected text includes table type.

Reply:
Key value may not be text. For example, the table of `UInt32` key.